src/app/login/page.tsx

```diff
@@ -4,27 +4,63 @@ import { useCallback, useRef } from "react";
 
 import { Button } from "@/components/ui/button";
 import { Input } from "@/components/ui/input";
+
 import { useNostrContext } from "@/lib/nostr/NostrContext";
+import { LoginType, fetchPublicKey, checkType } from "@/lib/utils";
 
 import { Puzzle } from "lucide-react";
 import Image from "next/image";
 import { useRouter } from "next/navigation";
 
+import { nip19 } from "nostr-tools"
+
+
 export default function Login() {
-  const { setAuthor } = useNostrContext();
+
+  const { setAuthor, addReplay } = useNostrContext();
   const router = useRouter();
+
   const inputRef = useRef<HTMLInputElement>(null);
 
-  /* try entering npub18zx8lw3947pghsgzqv2t0x8pe767sscag5djgj5afr755xkqd97qt530pr */
+  // TODO : setAuthor should be replaced in the future
```

Comment:
Why should setAuthor be replaced?

Reply:
![image](https://user-images.githubusercontent.com/120607606/224512620-417247dd-2bb1-4041-8616-20ca94341fc9.png)

`setAuthor()` is using npub, but all standards uses `hex` public key as reference, so we need to convert hex to `npub`  each time we call `setAuthor()`, something that could have been avoided by using nip-01 pubkey standard and changing to npub when needed

Reply:
I see, thanks for the clarification :+1: